evap/results/views.py

```diff
@@ -72,11 +72,14 @@ def course_detail(request, semester_id, course_id):
 
     sections = calculate_results(course)
 
+    public_view = request.GET.get('public_view', 'false') # default: show own view
+    public_view = {'true': True, 'false': False}.get(public_view.lower()) # convert parameter to boolean
+
     for section in sections:
         results = []
         for result in section.results:
             if isinstance(result, TextResult):
-                answers = [answer for answer in result.answers if user_can_see_text_answer(request.user, answer)]
+                answers = [answer for answer in result.answers if not public_view and user_can_see_text_answer(request.user, answer)]
```

Comment:
how about passing a parameter `public` into `user_can_see_text_answer` and returning false from there? it would be cool to keep the decision whether a comment is displayed or not in one place. but then it needs to have a new name; i didn't find one but maybe you do.